services/replay/service.go

```diff
@@ -766,6 +773,12 @@ func (s *Service) updateReplayResult(replay Replay, err error) {
 	}
 	replay.Progress = 1.0
 	replay.Date = time.Now()
+
+	r, err := s.replays.Get(replay.ID)
+	if err != nil {
+		s.diag.Error("failed to save replay results", err)
+	}
+	replay.ExecutionStats = r.ExecutionStats
```

Comment:
I think this logic is going to be confusing and cause issues later on.

Looking at the code I think a small refactor is what we need here. 

Let's make the `doReplay` function call the `updateReplayResult` function and have the `updateReplayResult` function take the stats as an argument. Then remove the calls to updateReplayResult outside of `doReplay`.
This will move ownership of updating the replay data in the store to one location so we don't have to worry about overriding previous changes when replacing the replay data.

Thoughts?